crates/matcher/src/line_splitter.rs

```diff
@@ -34,67 +35,90 @@ impl From<String> for LineSplitter {
     }
 }
 
+// XXX: avoid cloning everywhere below. Cloning is bad.
```

Comment:
You can try to delete all `clone()` for `item.display`. I don't see why types should not match here without `clone()`. With `item.filter` it is just `item.filter.as_deref()` and then you do whatever you want with `Option<&str>`.

Reply:
The borrow-check was complaining about partially moved `item`. I'll check what I can remove without making him angry.